macro/src/generate.rs

```diff
@@ -967,6 +976,116 @@ fn generate_to_primitive_traits(item: &BoundedInteger, tokens: &mut TokenStream)
     }
 }
 
+fn generate_index_traits(item: &BoundedInteger, tokens: &mut TokenStream) {
+    let ident = &item.ident;
+
+    tokens.extend(quote! {
+        impl<T> ::core::ops::Index<#ident> for [T] {
+            type Output = T;
+
+            #[inline]
+            fn index(&self, index: #ident) -> &Self::Output {
+                &self[index.get()]
+            }
+        }
+
+        impl<T ::core::ops::IndexMut<#ident> for [T] {
+            #[inline]
+            fn index_mut(&mut self, index: #ident) -> &mut Self::Output {
+                &mut self[index.get()]
+            }
+        }
+    });
+}
+
+fn generate_index_traits_alloc(item: &BoundedInteger, tokens: &mut TokenStream) {
```

Comment:
You can probably reduce code duplication here by having a single function like this:

```rs
fn generate_index_traits_alloc(item: &BoundedInteger, tokens: &mut TokenStream, alloc_or_std: Ident) {
```

Then you use `::#alloc_or_std::vec::Vec` in the function body, and parse it in like this:

```rs
if item.std {
    generate_index_traits_alloc(item, tokens, Ident::new("std", Span::call_site()));
} else {
    generate_index_traits_alloc(item, tokens, Ident::new("alloc", Span::call_site()));
}
```

Reply:
this still requires `else if item.alloc` right? If alloc is not present this fails otherwise

Reply:
Oh, you’re right yes :P